### PR TITLE
Added MultimodalModelGenerator

### PR DESCRIPTION
**Pull request by Izukimat**
## Summary

Adds a generic HuggingFace Multimodal Model Generator (HFMultimodalModelGenerator) to FedRAG, enabling unified inference for text, image, audio, and video-capable models via HuggingFace Transformers.

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [x] Code quality / linting
- [ ] Other (please describe):

## Description
Any information reviewers should be aware of:

This PR introduces HFMultimodalModelGenerator which supports multimodal chat inference for models like Gemma 3n, Llama 4, Mistral 3, and DeepSeek-VL.

Model class inference in _detect_model_class() is based on the presence of vision_config, audio_config, or video_config attributes in the model’s config, as there is no official HuggingFace API for automatic multimodal modality detection.

The generator expects the HuggingFace model output to have a .logits attribute for compute_target_sequence_proba(); if not present, a runtime error is raised.

The generator currently supports image, audio, and video modalities as separate arguments.

Implementation and interface are designed to be minimal, modular, and compatible with FedRAG’s base generator style.

Tested with Google’s Gemma3n model for image and audio inputs.

## Testing

Describe how you tested your changes. Include the steps to reproduce, commands run, and any relevant outputs.

- [x] Unit tests added or updated
- [x] All tests pass locally (`make test`)
- [x] Code coverage maintained or improved

## Checklist

Before submitting your PR, please check off the following:

- [x] My code follows the existing style and conventions
- [x] I’ve run linting (`make lint`)
- [x] I’ve added/updated relevant documentation
- [x] I’ve added/updated tests as needed
- [ ] I’ve verified integration with existing tools (HuggingFace, LlamaIndex, LangChain, etc. if applicable)
- [ ] I’ve added an entry to the CHANGELOG.md (if applicable)

## Related Issues or PRs

If this PR addresses or relates to existing issues or pull requests, link them here:

- Closes #
- Related to #

### CODE REVIEW

File: src/fed_rag/generators/huggingface/hf_multimodal_model.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+            add_generation_prompt=False,
+            tokenize=True,
+            return_dict=True,
+            return_tensors="pt",
+        )
+        prompt_len = prompt_inputs["input_ids"].shape[-1]
+        with torch.no_grad():
+            outputs = self.model(**inputs)
+        if not hasattr(outputs, "logits") or outputs.logits is None:
+            raise RuntimeError(
```
Comment (nerdai) [inline src/fed_rag/generators/huggingface/hf_multimodal_model.py:249]: Can we raise a GeneratorError here

File: src/fed_rag/generators/huggingface/hf_multimodal_model.py [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        return torch.exp(torch.tensor(sum(log_probs)))
+
+    @property
+    def model(self) -> PreTrainedModel:
+        if self._model is None:
+            self._model = self._load_model_from_hf()
+        return self._model
+
+    @property
+    def tokenizer(self) -> BaseTokenizer:
```
Comment (nerdai) [inline src/fed_rag/generators/huggingface/hf_multimodal_model.py:268]: Is Processor a generalization of tokenizer? It looks like it has a similar interface. Should we create a processor property as well for syntactic sugar for this multimodal generator?

Comment (nerdai) [reply src/fed_rag/generators/huggingface/hf_multimodal_model.py:268]: Nevermind-I see you have one. If it's a generalization of tokenizer let's keep it. If these two are conceptually different, I wonder if we need to raise warning or just raise not implemented or attribute errror

**Review by nerdai — COMMENTED** [review]
Looking good! I just have one primary request which is to handle imports of HF similar to how we do it for other generators. That is, we can import module even if the HF extra is not installed. Users will get an error at initialization if the error is missing.

File: src/fed_rag/generators/huggingface/hf_multimodal_model.py (hunk 2 above)
Comment (Izukimat) [reply src/fed_rag/generators/huggingface/hf_multimodal_model.py:268]: Right, that sounds reasonable. I will raise AttributeError if no tokenizer is callable.

**Review by nerdai — COMMENTED** [review]
Nice! Sorry just one last thing then we should be good to go. Can you see the generate method -- it's overloading the base class method. In other words it breaks the interface.

File: src/fed_rag/generators/huggingface/hf_multimodal_model.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+            queries = [self.to_query(p) for p in prompt]
+        else:
+            queries = self.to_query(prompt)
+        return self.generate(query=queries, context=None, **kwargs)
+
+    def compute_target_sequence_proba(
+        self,
+        prompt: Prompt | str,
+        target: str,
+        context: Context | str | None = None,
```
Comment (nerdai) [inline src/fed_rag/generators/huggingface/hf_multimodal_model.py:199]: Does this break the interface of super class as well?

**Review by nerdai — COMMENTED** [review]
Found another one where were overloading the super method with additional params...
